authz/policy/common/src/main/java/org/keycloak/authorization/policy/provider/permission/ScopePolicyProviderFactory.java

```diff
@@ -62,6 +66,33 @@ public ScopePermissionRepresentation toRepresentation(Policy policy, Authorizati
         return new ScopePermissionRepresentation();
     }
 
+    @Override
+    public void onCreate(Policy policy, ScopePermissionRepresentation representation, AuthorizationProvider authorization) {
```

Comment:
In addition to `onCreate|Update` you should also make sure the representation is set with the `defaultResorceType` setting by implementing `org.keycloak.authorization.policy.provider.permission.ScopePolicyProviderFactory#toRepresentation`.  

Please, take a look at `ResourcePolicyProviderFactory`.

Reply:
Done!